doc/contributing.md

```diff
@@ -0,0 +1,231 @@
+# Contributing Guide
+This guide targets new members looking for a guide on contributing to the
+project. This guide will focus on the overall picture of effective coding and
+contributions. It is not intended to be a technical guide.
+
+## Initial Notes
+Contributing to open source presents a unique set of challenges, especially when
+dealing with large projects. Our project is approximately 50-80k lines of code,
+and is still very small by some open source standards. 
+
+This should not be discouraging, but rather informative. Most initial
+contributions will be small; it takes time to learn the project structure and any
+languages and skills each particular project requires. Contributions of any size
+will be greatly appreciated, and you should have realistic expectations of what
+a first contribution might be (if you dont believe me, you can look at my first
+contribution [here](https://github.com/RoboJackets/robocup-software/commit/6ce98fc0f8d88b6d145700779e126c0f1b99bb92#diff-71a3477f37bd5b20744e292eda2e3fbc). It's two lines of 'code'). This guide will help you learn to navigate our code base, and 
+work toward an initial contribution.
+
+## Initial Prerequisites
+Before starting, you'll need a Unix-like environment. This means you need to be 
+running OSX or Ubuntu Linux (other flavors of Debian may work, but we do not
+officially support them or FedoraCore).
+
+You'll also need a Github account, which you can create [here](http://github.com).
+Github is a web front-end for a program called Git, which allows multiple people to
+work on and contrubute to the same code base, at the same time.
+
+## Git
+We use Git as our version control system (if you already know git, you can skip
+ahead). A Version control system allows many people to code for the same project 
+at the same time.
+
+To ease new contributors into Git, I'll repeatedly use the analogy of the 
+classroom test.
+
+If you simply want to learn about the workflow we use, and are less interested
+in learning the relationships between Git elements, you can skip to the "Overall
+Workflow" section.
+
+### Master
+If Git is like a test, then you can think of the master branch as the final copy
+you submit for grading. This copy should have the correct answer, should contain
+the most effective or efficient solution, and should be highly neat and readable
+for the graders. You probably shouldn't do any work here, but rather should
+explore in other locations such as scrap paper. Once you have several solutions,
+you can pick the one you like the best, and more neatly copy the work on to the
+test you will hand in.
+
+Our master branch can be found [here](https://github.com/RoboJackets/robocup-software).
+The code in master is always neat and untouched. It can always compile. When you
```

Comment:
Maybe we should say 'tip of master' or 'latest master', we have way too many commits that don't compile gone in there...

On that note, thats a slightly bad thing, because that makes git bisects really hard... :frowning: